clang/lib/Sema/SemaCoroutine.cpp

```diff
@@ -1198,6 +1198,11 @@ void Sema::CheckCompletedCoroutineBody(FunctionDecl *FD, Stmt *&Body) {
   if (FD->hasAttr<AlwaysInlineAttr>())
     Diag(FD->getLocation(), diag::warn_always_inline_coroutine);
 
+  // We don't allow use of VLAs within a coroutine, so diagnose if we've seen
```

Comment:
```suggestion
  // VLAs are not allowed within a coroutine, so diagnose if we've seen
```

nit: it is not us (clang/LLVM) who decide not to support VLA in coroutines. But C++20 coroutines itself doesn't support VLA by its design naturally.